lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -56,6 +56,77 @@ removeNode = (node) ->
   removedNode = node.parentNode.removeChild(node)
   triggerEvent('page:after-node-removed', removedNode)
 
+updateHead = (doc, callback) ->
+  loadedAssets = extractTrackedAssets document
+  fetchedAssets = extractTrackedAssets doc
+
+  updateLinkTags(loadedAssets, fetchedAssets)
+  updateScriptTags(loadedAssets, fetchedAssets, callback)
+
+extractTrackedAssets = (doc) ->
+  for node in doc.head.childNodes when node.getAttribute?('data-turbolinks-track')?
+    node
+
+insertScriptTask = (scriptNode) ->
+  newNode = document.createElement('SCRIPT')
+  newNode.setAttribute attr.name, attr.value for attr in scriptNode.attributes
+  newNode.appendChild document.createTextNode scriptNode.innerHTML
+
+  return (done) ->
+    onScriptEvent = ->
+      newNode.removeEventListener('load', onScriptEvent)
+      newNode.removeEventListener('error', onScriptEvent)
+      done()
+    newNode.addEventListener('load', onScriptEvent)
+    newNode.addEventListener('error', onScriptEvent)
+    document.head.appendChild(newNode)
+
+filterForNodeType = (nodeType) ->
+  (node) -> node.nodeName == nodeType
+
+asyncEach = (tasks, callback) ->
+  callback() if tasks.length == 0
+  tasksRemaining = tasks.length
+  done = ->
+    tasksRemaining--
+    if tasksRemaining == 0
+      callback()
+    return
+
+  tasks.forEach (task) ->
+    task done
+    return
+  return
+
+updateScriptTags = (loadedAssets, fetchedAssets, callback) ->
+  oldScripts = loadedAssets.filter(filterForNodeType('SCRIPT'))
+  newScripts = fetchedAssets.filter(filterForNodeType('SCRIPT'))
+  nodeIsNew = (node) ->
+    !(oldScripts.some((listNode) -> node.isEqualNode(listNode)))
+
+  scriptTasks = newScripts
+    .filter(nodeIsNew)
+    .map(insertScriptTask)
+
+  asyncEach(scriptTasks, callback)
```

Comment:
Well, we could rebuild `asyncEach` to not execute a function until the previous one's `done` is called. That would obviously take more time to finish, but if it makes things not break maybe it's the right call?